orchestration/src/test/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatModelTest.java

```diff
@@ -67,5 +97,66 @@ void testThrowsOnMissingLlmConfig() {
         .isExactlyInstanceOf(IllegalStateException.class)
         .hasMessageContaining("LLM config is required");
+    assertThatThrownBy(() -> client.stream(new Prompt("test", emptyConfig)))
+        .isExactlyInstanceOf(IllegalStateException.class)
+        .hasMessageContaining("LLM config is required");
+  }
+
+  @Test
+  void streamChatCompletionOutputFilterErrorHandling() throws IOException {
+    try (var inputStream = spy(fileLoader.apply("streamChatCompletionOutputFilter.txt"))) {
+
+      final var httpClient = mock(HttpClient.class);
+      ApacheHttpClient5Accessor.setHttpClientFactory(destination -> httpClient);
+
+      // Create a mock response
+      final var mockResponse = new BasicClassicHttpResponse(200, "OK");
+      final var inputStreamEntity = new InputStreamEntity(inputStream, ContentType.TEXT_PLAIN);
+      mockResponse.setEntity(inputStreamEntity);
+      mockResponse.setHeader("Content-Type", "text/event-stream");
+
+      // Configure the HttpClient mock to return the mock response
+      doReturn(mockResponse).when(httpClient).executeOpen(any(), any(), any());
+
+      Flux<ChatResponse> flux = client.stream(prompt);
+      assertThatThrownBy(() -> flux.toStream().forEach(System.out::println))
+          .isInstanceOf(OrchestrationClientException.class)
+          .hasMessage("Content filter filtered the output.");
+
+      Mockito.verify(inputStream, times(1)).close();
+    }
+  }
+
+  @Test
+  void testStreamCompletion() throws IOException {
+    try (var inputStream = spy(fileLoader.apply("streamChatCompletion.txt"))) {
+
+      final var httpClient = mock(HttpClient.class);
+      ApacheHttpClient5Accessor.setHttpClientFactory(destination -> httpClient);
+
+      // Create a mock response
+      final var mockResponse = new BasicClassicHttpResponse(200, "OK");
+      final var inputStreamEntity = new InputStreamEntity(inputStream, ContentType.TEXT_PLAIN);
+      mockResponse.setEntity(inputStreamEntity);
+      mockResponse.setHeader("Content-Type", "text/event-flux");
+
+      // Configure the HttpClient mock to return the mock response
+      doReturn(mockResponse).when(httpClient).executeOpen(any(), any(), any());
+
+      Flux<ChatResponse> flux = client.stream(prompt);
+      var deltaList = flux.toStream().toList();
+
+      assertThat(deltaList).hasSize(3);
+      // the first delta doesn't have any content
+      assertThat(deltaList.get(0).getResult().getOutput().getContent()).isEqualTo("");
+      assertThat(deltaList.get(1).getResult().getOutput().getContent()).isEqualTo("Sure");
+      assertThat(deltaList.get(2).getResult().getOutput().getContent()).isEqualTo("!");
+
+      assertThat(deltaList.get(1).getResult().getMetadata().getFinishReason()).isEqualTo("");
+      assertThat(deltaList.get(1).getResult().getMetadata().getFinishReason()).isEqualTo("");
+      assertThat(deltaList.get(2).getResult().getMetadata().getFinishReason()).isEqualTo("stop");
```

Comment:
(Question)

Wouldn't it be `get(3)` that returns finish reason stop?

Reply:
The third line (n2) is the stop